src/raftstore/store/worker/apply.rs

```diff
@@ -1228,62 +1307,73 @@ impl ApplyDelegate {
             .with_label_values(&["batch-split", "success"])
             .inc();
 
-        Ok((resp, Some(ExecResult::SplitRegion { regions, derived })))
+        future::ok((
+            self,
+            ctx,
+            resp,
+            Some(ExecResult::SplitRegion { regions, derived }),
+        ))
     }
 
     fn exec_prepare_merge(
-        &mut self,
```

Comment:
Seems many of these functions have been changed to consume `self`, and they seem to return `Future<Item = (Self, _), Error = (Self, _)>`. It seems a bit awkward. Was it not possible to still use a `&mut self`?

Reply:
Each stage maybe running in different thread, so we have to use move for delegate.